### PR TITLE
[Docs][ci] Update the documentation about buildkite

### PR DESCRIPTION
**Pull request by jsanz**
This finishes the migration to Buildkite with some documentation on how the build and deployment to staging and production works.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,32 +1,34 @@
...
 ## New Releases
+
 New releases of EMS Landing Page match minor releases of the Elastic Stack.
 
 To add a new release:
 1. Open a PR with the following changes:
-    1. Create a new config in the [.ci/jobs](https://github.com/elastic/ems-landing-page/tree/master/.ci/jobs) directory for a new release branch.
     1. Change the `EMS_VERSION` in config.json.
-    1. Upgrade the `@elastic/ems-client` dependency.
-    1. Bump the verison in `package.json`.
```
Comment (nickpeihl) [inline CONTRIBUTING.md:21]: 😑  nice catch!

**Review by nickpeihl — APPROVED** [review]
lgtm! thanks!

**Comment by jsanz** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|v8.9|[](https://github.com/elastic/ems-landing-page/pull/605)|
|✅|v8.8|[](https://github.com/elastic/ems-landing-page/pull/606)|
|✅|v7.17|[](https://github.com/elastic/ems-landing-page/pull/607)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**